src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -915,5 +918,82 @@ void AppHost::_HandleSettingsChanged(const winrt::Windows::Foundation::IInspecta
 void AppHost::_IsQuakeWindowChanged(const winrt::Windows::Foundation::IInspectable&,
                                     const winrt::Windows::Foundation::IInspectable&)
 {
+    if (_window->IsQuakeWindow() && !_logic.IsQuakeWindow())
+    {
+        // If we're exiting quake mode, we should make our
+        // tray icon disappear.
+        if (_trayIconData)
+        {
+            Shell_NotifyIcon(NIM_DELETE, &_trayIconData.value());
+            _trayIconData.reset();
+        }
+    }

```

Comment:
**discussion**: should we 

```c++
	else if (!_window->IsQuakeWindow() && _logic.IsQuakeWindow())
	{
        _UpdateTrayIcon();
    }
```

to show the icon as soon as the quake window is created? Or do we want the icon to only appear when the window is minimized?

Reply:
You know what, I like that. The icon will kinda serve as an indicator that a quake window is alive (hidden or not), and the user can always rely on that icon to get focus the quake window.

In fact, right now the icon is still visible when the quake window is visible. So, it's actually a bit inconsistent because the _first_ time you bring up the quake window, the icon isn't there yet (because it'll only add it on a minimize). So yeah I think I'll add the icon as soon as the window's created.

I guess another route that's possible is to only show the icon when the quake window is minimized, and if it's not (aka. shown/restored), remove the icon? Still kinda like having the icon there in all states though.

Reply:
> Still kinda like having the icon there in all states though. 

me too

Reply:
I haven't looked through the spec in a while, but wasn't the plan to have the tray icon serve as a way to target windows with specific names and stuff like that? Like, if the tray icon is supposed to provide functionality outside of quake mode, shouldn't we always show it?

Reply:
[feature today] -> [feature tomorrow] -> [feature in 1 year] :smile:

Reply:
I think in this iteration of the tray icon (which only works for quake mode), it makes sense to me to show it only if there's a quake window alive. Once I've gotten more of the minimize to tray/tray icon features down that warrant the "always show" behavior, it should be fairly simple to make that change.